client/tests/client/client_test.py

```diff
@@ -37,12 +38,13 @@ async def test_hub_flow(
     # Watch the progress meter
     progress = configured_client.watch_spawn_progress()
     progress_pct = -1
-    async with asyncio.timeout(30):
-        async for message in progress:
-            if message.ready:
-                break
-            assert message.progress > progress_pct
-            progress_pct = message.progress
+    async with aclosing(progress):
+        async with asyncio.timeout(30):
```

Comment:
Just curious if the `aclosing` has to be outside the `timeout`, or if it doesn't matter.

Reply:
It shouldn't matter. It will close the generator in the `finally` block when exited by the exception raised by `asyncio.timeout` either way.